content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -218,204 +218,173 @@ this variable to checkout the appropriate version of the library.
+Use a static class or instantiate a local variable of a class instead.
 
-Internally, scripts in the `vars` directory are instantiated on-demand  as
-singletons. This allows multiple methods or properties to be defined in a
-single `.groovy` file which interact with each other, for example:
+The example below shows a global variable `customer` which stores a `name` value.
+It also outputs a message when name is set.
 
+.var/customer.groovy
 [source,groovy]
 ----
-// vars/acme.groovy
 def setName(value) {
-    name = value
+    privateName = value
+    echo "Setting name to ${privateName}"
 }
 def getName() {
-    name
-}
-def caution(message) {
-    echo "Hello, ${name}! CAUTION: ${message}"
-}
-----
-
-In the above, `name` is not referring to a field (even if you write it as `this.name`!),
-but to an entry created on demand in a `Script.binding`.
-To be clear about what data you intend to store and of what type,
-you can instead provide an explicit class declaration
-(the class name should match the file basename):
-
-[source,groovy]
-----
-// vars/acme.groovy
-class acme implements Serializable {
-    private String name
-    def setName(value) {
-        name = value
-    }
-    def getName() {
-        name
-    }
-    def caution(message) {
-        echo "Hello, ${name}! CAUTION: ${message}"
-    }
+    privateName
 }
 ----
 
-The Pipeline can then invoke these methods which will be defined on the
-`acme` object:
-
-[source,groovy]
-----
-acme.name = 'Alice'
-echo acme.name /* prints: 'Alice' */
-acme.caution 'The queen is angry!' /* prints: 'Hello, Alice. CAUTION: The queen is angry!' */
-----
-
-[NOTE]
-====
-A variable defined in a shared library will only show up in _Global Variables
-Reference_ (under _Pipeline Syntax_) after Jenkins loads and uses that library
-as part of a successful Pipeline run.
 ====
 
-=== Defining steps
+=== Custom steps
 
-Shared Libraries can also define global variables which behave similarly to
-built-in steps, such as `sh` or `git`. Global variables defined in Shared
-Libraries *must* be named with all lower-case or "camelCased" in order to be
-loaded properly by Pipeline.
-footnote:[https://gist.github.com/rtyler/e5e57f075af381fce4ed3ae57aa1f0c2]
+Custom steps are a special form of Global Variable,
+which create a method with the same name as their containing file
+(instead of an object with one or more methods attached).  They behave similar to
+built-in steps, such as `sh` or `git`, and can also call built-in steps, other
+other custom steps, or other valid Pipeline
+link:http://groovy-lang.org/syntax.html[Groovy code].
 
-For example, to define `sayHello`, the file `vars/sayHello.groovy`
-should be created and should implement a `call` method. The `call` method
-allows the global variable to be invoked in a manner similar to a step:
+To define a custom step create a file in the `vars` directory containing only a `call` method.
```

Comment:
"only"? Are helper methods not allowed? If they are, maybe write "[...]containing [] a `call` method."